clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java

```diff
@@ -321,18 +330,51 @@ public void close() {}
         // for a normal iterator, we cannot ensure that the underlying compression stream is closed,
         // so we decompress the full record set here. Use cases which call for a lower memory footprint
         // can use `streamingIterator` at the cost of additional complexity
-        try (CloseableIterator<Record> iterator = compressedIterator(BufferSupplier.NO_CACHING)) {
+        try (CloseableIterator<Record> iterator = compressedIterator(BufferSupplier.NO_CACHING, false)) {
             List<Record> records = new ArrayList<>(count());
             while (iterator.hasNext())
                 records.add(iterator.next());
             return records.iterator();
         }
     }
 
+    @Override
+    public CloseableIterator<Record> skipKeyValueIterator() {
+        if (count() == 0) {
+            Iterator<Record> inner = Collections.emptyIterator();
+            return new CloseableIterator<Record>() {
+                @Override
+                public void close() {}
```

Comment:
Out of curiosity, when do we choose to put curly braces in one line, and we choose to put them in two lines?

Reply:
Most of the time we just put them in one line; if there's some comments explaining why it's a no-op it would be multi-lines.